assets/js/components/activation/activation-app.js

```diff
@@ -43,15 +43,19 @@ export class ActivationApp extends Component {
 			buttonLabel = __( 'Go to Dashboard', 'google-site-kit' );
 		}
 
+		const onButtonClick = async ( event ) => {
```

Comment:
Ideally this would use `useCallback` to prevent `ActivationMain` from rerendering each time this component renders because `onButtonClick` will be different. Looks like this component is basically a functional component already so this would be a very straightforward refactor.

As a class component, I believe the alternative is to add a constructor and add the callback as a class method.
E.g.
```js
constructor( props ) {
    super( props );
    this.onButtonClick = this.onButtonClick.bind( this );
}
```
Not a big deal in this case since this component probably doesn't render often but changing it to be a functional component is trivial.